src/main/java/org/eclipse/yasson/internal/ConstructorPropertiesAnnotationIntrospector.java

```diff
@@ -0,0 +1,73 @@
+package org.eclipse.yasson.internal;
+
+import org.eclipse.yasson.internal.components.JsonbComponentInstanceCreator;
+import org.eclipse.yasson.internal.model.CreatorModel;
+import org.eclipse.yasson.internal.model.JsonbCreator;
+import org.eclipse.yasson.internal.properties.MessageKeys;
+import org.eclipse.yasson.internal.properties.Messages;
+
+import java.beans.ConstructorProperties;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Parameter;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.logging.Logger;
+
+class ConstructorPropertiesAnnotationIntrospector {
+
+    private static final Logger LOG = Logger.getLogger(JsonbComponentInstanceCreator.class.getName());
+
+    private final JsonbContext jsonbContext;
+    private final AnnotationFinder<ConstructorProperties> constructorProperties;
+
+    public static final ConstructorPropertiesAnnotationIntrospector forContext(JsonbContext jsonbContext) {
+	return new ConstructorPropertiesAnnotationIntrospector(jsonbContext, AnnotationFinder.findAnnotation(ConstructorProperties.class));
+    }
+
+    protected ConstructorPropertiesAnnotationIntrospector(JsonbContext context, AnnotationFinder<ConstructorProperties> annotationFinder) {
+	this.jsonbContext = context;
+	this.constructorProperties = annotationFinder;
+    }
+
+    public JsonbCreator getCreator(Class<?> clazz) {
+	JsonbCreator jsonbCreator = null;
+
+	for (Constructor<?> constructor : declaredConstructorsIn(clazz)) {
```

Comment:
This code calls and iterates over `Class#getDeclaredConstructors` for the second time unnecessary. While it may be not a major performance issue, it is better to avoid it. `AnnotaionIntrospector` has already similar iteration in `getCreator()` method, it would be better to either: do it right there, or at least cache the `Class#getDeclaredConstructors` to avoid calling it for the second time.

Reply:
I'll use the array of constructors instead of the class as parameter to avoid getting them a second time.